bitriseclient/bitriseclient.go

```diff
@@ -0,0 +1,939 @@
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+
+		logDebugPretty(requestResponse)
+		return nil
+
+	}); err != nil {
+		return err
+	}
+
+	// Success
+	log.Donef("Request succeeded with status code: %d", responseStatusCode)
+
+	return nil
+}
+
+// -------------------------------------------------
+// -- Certificates
+
+// FetchUploadedIdentities ...
+func (client *BitriseClient) FetchUploadedIdentities() ([]FetchUploadedIdentityListResponseData, error) {
+	log.Debugf("\nDownloading provisioning profile list from Bitrise...")
+
+	requestURL, err := urlutil.Join(baseURL, appsEndPoint, client.selectedAppSlug, certificatesEndPoint)
+	if err != nil {
+		return []FetchUploadedIdentityListResponseData{}, err
+	}
+
+	log.Debugf("\nRequest URL: %s", requestURL)
+
+	headers := map[string]string{
+		"Authorization": "token " + client.accessToken,
+	}
+
+	request, err := createRequest(http.MethodGet, requestURL, headers, map[string]interface{}{})
+	if err != nil {
+		return []FetchUploadedIdentityListResponseData{}, err
+	}
+
+	// Response struct
+	requestResponse := FetchUploadedIdentityListResponse{}
+	responseStatusCode := -1
+
+	//
+	// Perform request
+	if err := retry.Times(1).Wait(5 * time.Second).Try(func(attempt uint) error {
+		body, statusCode, err := performRequest(request)
+		if err != nil {
+			log.Warnf("Attempt (%d) failed, error: %s", attempt+1, err)
+			if !strings.Contains(err.Error(), "failed to perform request") {
+				log.Warnf("Response status: %d", statusCode)
+				log.Warnf("Body: %s", string(body))
+			}
+			return err
+		}
+
+		responseStatusCode = statusCode
+
+		// Parse JSON body
+		if err := json.Unmarshal([]byte(body), &requestResponse); err != nil {
+			return fmt.Errorf("failed to unmarshal response (%s), error: %s", body, err)
+		}
+		return nil
+
+	}); err != nil {
+		log.Errorf("Fetching list failed %s", err)
+		return []FetchUploadedIdentityListResponseData{}, err
+	}
+
+	// Success
+	log.Debugf("Request succeeded with status code: %d", responseStatusCode)
+	logDebugPretty(requestResponse)
+
+	return requestResponse.Data, nil
+}
+
+// GetUploadedCertificatesSerialby ...
+func (client *BitriseClient) GetUploadedCertificatesSerialby(identitySlug string) (certificateSerialList []big.Int, err error) {
+	downloadURL, certificatePassword, err := client.getUploadedIdentityDownloadURLBy(identitySlug)
+	if err != nil {
+		return []big.Int{}, err
+	}
+
+	content, err := client.downloadUploadedIdentity(downloadURL)
+	if err != nil {
+		return []big.Int{}, err
+	}
+
+	certificates, err := certificateutil.CertificatesFromPKCS12Content([]byte(content), certificatePassword)
+	if err != nil {
+		return []big.Int{}, err
+	}
+
+	serialList := []big.Int{}
```

Comment:
if the intial value does not matter use: var serialList  []big.Int
serialList := []big.Int{}